streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -52,6 +52,7 @@
     final Logger log;
     final LogContext logContext;
     final StateDirectory stateDirectory;
+    final boolean hasPersistentLocalStore;
```

Comment:
Do we need this? Might be simpler to update `StateDirectory` to not acquire the `lock()` but just return `true` for this case?

Reply:
Assume that there are multiple tasks in a topology in which some of them have persistent state stores and others have in-memory stores. In this case, `StateDirectory` don't have any knowledge whether to create the `TaskId` directory or not.

Reply:
Maybe I am missing something, but if there is no stateful task, we don't create the `StateDirectory ` -- and task directories are nested within `StateDirectory`:
```
StateDirectory == /<state.dir/<application.id>/

overall:
/<state.dir>/<application.id>/<task.id>/<store>
```
From my understanding, the ticket addresses the case that there is no disk. If there is disk, and there is one stateful task, we can also create task directories from stateless task -- I agree that it would be ok to not create them, but this increase code complexity and the gain seems to be small?

Thoughts?

Reply:
@mjsax I was thinking the same thing.  I understand the goal of not creating the directories for stateless tasks, but I'm not sure the increase in code buys much.

Reply:
Updated the code to create task directory for all the tasks even if any one of the task has persistent store.